pyne/vta.py

```diff
@@ -0,0 +1,500 @@
+"""Purpose:
+
+This module provides a modified ray traverse method, which is modifed based on
+http://www.cse.yorku.ca/~amana/research/grid.pdf, to calculate which voxels
+intersect with a given ray (line segment).
+Several rays are used to calculate the voxels that intersect with the facet.
+A series of rays are constructed based on both the facet and the mesh grid to
+make sure all the wanted voxels will be found. Combining all the found voxels
+for each ray, we get the voxels intersecting with the facet.
+
+These voxels will be used to construct the bounding volume for source sampling.
+But currently these functions are separate from MOAB. Interface of MOAB with
+PyNE is still in need.
+
+.. moduleauthor:: X. Zhang
+"""
+from __future__ import division
+import math
+import numpy as np
+
+
+##################################
+### Facet voxel traverse tools ###
+##################################
+
+
+def _is_point_in_mesh(bounds, point):
+    """
+    Check whether a point is in the mesh.
+
+    Parameters:
+    -----------
+    bounds : list
+        Boundaries. Nested list of the boundaries along x, y, and z directions.
+        Expected format: [[x_min, ..., x_max],
+                          [y_min, ..., y_max],
+                          [z_min, ..., z_max]]
+    point : numpy array
+        An array with size of 3, represents the point to be checked.
+
+    Return:
+    -------
+    True : when point in the mesh.
+    False : when point outside the mesh.
+    """
+    if bounds[0][0] <= point[0] <= bounds[0][-1] and \
+       bounds[1][0] <= point[1] <= bounds[1][-1] and \
+       bounds[2][0] <= point[2] <= bounds[2][-1]:
+        return True
+    else:
+        return False
+
+def _find_voxel_idx_1d(bounds_1d, cor, vec_1d):
+    """
+    Find the voxel id (in a given dimension).
+
+    Parameters:
+    -----------
+    bounds_1d : list
+        Boundary of the mesh in a given direction.
+    cor : float
+        Coordinate of the point in the given direction.
+    vec_1d : float
+        Vector attribute in the given direction. Used for determining the
+        search direction.
+
+    Return:
+    -------
+    idx : int
+        Mesh index of the given direction. This value is -1 when the point not
+        in the mesh.
+    ----------
+    """
+    idx = -1
+    bounds_1d = np.array(bounds_1d)
```

Comment:
What is the format of bounds_1d?. Is it from the nested list you mentioned in "_is_point_in_mesh" function?. If so then you may consider making the nested list into an array with each row containing the boundaries along one dimension;
bounds = np.array([[x_min, ..., x_max],
                                [y_min, ..., y_max],
                                [z_min, ..., z_max]]) 
This way bounds_1d = bounds[0] (or [1] or [2] according yo which direction is considered) and it is a numpy array so you wouldn't need to cast bounds_1d as an array here.